bubblewrap.c

```diff
@@ -3143,12 +3143,12 @@ main (int    argc,
   if (pid != 0)
     {
       /* Parent, outside sandbox, privileged (initially) */
+      cleanup_fdp (&intermediate_pids_sockets[PIPE_WRITE_END]);
```

Comment:
What about assigning `-1` to the passe pointer in `cleanup_fdp()`, so a potential second `close()` call won't close an uninvolved descriptor.

For the cleanup attribute annotated variables compilers *should* be able to optimize the assignment out.

Another unrelated thought: systemd uses a `safe_close()` wrapper instead of directly `close(3)` to assert `EBADF` was not returned, to spot file descriptor mixups, see https://github.com/systemd/systemd/blob/v256.7/src/basic/fd-util.c#L59.

Reply:
> What about assigning -1 to the passe pointer in cleanup_fdp()

Oh... I thought it already did that, but in fact it doesn't, so this PR needs redoing with either a different function or an augmented version of `cleanup_fdp()` (as do #666 and #667).

Reply:
> This closes the fd and sets the variable to -1 as a single operation

[citation needed]